File: web/src/model/victron_controller/dashboard/from_0_2_0_command.ts
// @ts-nocheck
import {Command as v0_2_0_Command, SetBoolKnob as v0_2_0_SetBoolKnob, SetFloatKnob as v0_2_0_SetFloatKnob, SetUintKnob as v0_2_0_SetUintKnob, SetDischargeTime as v0_2_0_SetDischargeTime, SetDebugFullCharge as v0_2_0_SetDebugFullCharge, SetForecastDisagreementStrategy as v0_2_0_SetForecastDisagreementStrategy, SetChargeBatteryExtendedMode as v0_2_0_SetChargeBatteryExtendedMode, SetExtendedChargeMode as v0_2_0_SetExtendedChargeMode, SetMode as v0_2_0_SetMode, SetKillSwitch as v0_2_0_SetKillSwitch, SetBookkeeping as v0_2_0_SetBookkeeping} from './v0_2_0/Command'
import {Command as dashboard_Command, SetBoolKnob as dashboard_SetBoolKnob, SetFloatKnob as dashboard_SetFloatKnob, SetUintKnob as dashboard_SetUintKnob, SetDischargeTime as dashboard_SetDischargeTime, SetDebugFullCharge as dashboard_SetDebugFullCharge, SetForecastDisagreementStrategy as dashboard_SetForecastDisagreementStrategy, SetChargeBatteryExtendedMode as dashboard_SetChargeBatteryExtendedMode, SetExtendedChargeMode as dashboard_SetExtendedChargeMode, SetMode as dashboard_SetMode, SetKillSwitch as dashboard_SetKillSwitch, SetBookkeeping as dashboard_SetBookkeeping} from './Command'

export function convert__command__from__0_2_0(from: v0_2_0_Command): dashboard_Command {
    if (from instanceof v0_2_0_SetBoolKnob) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetBoolKnob
    }
    if (from instanceof v0_2_0_SetFloatKnob) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetFloatKnob
    }
    if (from instanceof v0_2_0_SetUintKnob) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetUintKnob
    }
    if (from instanceof v0_2_0_SetDischargeTime) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetDischargeTime
    }
    if (from instanceof v0_2_0_SetDebugFullCharge) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetDebugFullCharge
    }
    if (from instanceof v0_2_0_SetForecastDisagreementStrategy) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetForecastDisagreementStrategy
    }
    if (from instanceof v0_2_0_SetChargeBatteryExtendedMode) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetChargeBatteryExtendedMode
    }
    if (from instanceof v0_2_0_SetExtendedChargeMode) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetExtendedChargeMode
    }
    if (from instanceof v0_2_0_SetMode) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetMode
    }
    if (from instanceof v0_2_0_SetKillSwitch) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetKillSwitch
    }
    if (from instanceof v0_2_0_SetBookkeeping) {
        return JSON.parse(JSON.stringify(from)) as dashboard_SetBookkeeping
    }

    throw new Error("Unknown ADT branch: " + from);
}

File: crates/dashboard-model/src/victron_controller/dashboard/from_0_2_0_command.rs


pub fn convert__command__from__0_2_0(from: &crate::victron_controller::dashboard::v0_2_0::command::Command) -> crate::victron_controller::dashboard::command::Command {
    match from {
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetBoolKnob(x) => crate::victron_controller::dashboard::command::Command::SetBoolKnob(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetFloatKnob(x) => crate::victron_controller::dashboard::command::Command::SetFloatKnob(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetUintKnob(x) => crate::victron_controller::dashboard::command::Command::SetUintKnob(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetDischargeTime(x) => crate::victron_controller::dashboard::command::Command::SetDischargeTime(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetDebugFullCharge(x) => crate::victron_controller::dashboard::command::Command::SetDebugFullCharge(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetForecastDisagreementStrategy(x) => crate::victron_controller::dashboard::command::Command::SetForecastDisagreementStrategy(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetChargeBatteryExtendedMode(x) => crate::victron_controller::dashboard::command::Command::SetChargeBatteryExtendedMode(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetExtendedChargeMode(x) => crate::victron_controller::dashboard::command::Command::SetExtendedChargeMode(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetMode(x) => crate::victron_controller::dashboard::command::Command::SetMode(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetKillSwitch(x) => crate::victron_controller::dashboard::command::Command::SetKillSwitch(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
        crate::victron_controller::dashboard::v0_2_0::command::Command::SetBookkeeping(x) => crate::victron_controller::dashboard::command::Command::SetBookkeeping(serde_json::from_value(serde_json::to_value(x).unwrap()).unwrap()),
    }
}